tools/toolchain/image

```diff
@@ -1 +1 @@
-docker.io/scylladb/scylla-toolchain:fedora-38-20240321
+docker.io/scylladb/scylla-toolchain:fedora-38-20240506
```

Comment:
@xtrey please update the image to `docker.io/scylladb/scylla-toolchain:fedora-38-20240510`

It will work on both arch

![image](https://github.com/scylladb/scylladb/assets/9914527/17677bb8-5918-4541-8ea9-4ad0229a3e34)